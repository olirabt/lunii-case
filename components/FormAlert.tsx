import cn from "classnames";
import Link from "next/link"

interface FormAlertProps {
	type: "success" | "warning";
	originalUrl: string;
	shortUrl: string;
	error: string | null;
}

const FormAlert: React.FC<FormAlertProps> = ({ type, originalUrl = "", shortUrl = "", error = "" }) => {

	const alertClass = cn({
		"bg-validate500 text-validate400": type === "success",
		"bg-alert400 text-alert500": type === "warning",
	});

	const RedirectLink = <Link href={originalUrl}>{shortUrl}</Link>;

	return (
		<div className={cn("text-left w-full px-5 py-3 mx-auto rounded-md", alertClass)}>
			{type === "success" && <p>Voici le lien raccourci : {RedirectLink}</p>}
			{type === "warning" && <p>{error}</p>}
		</div>
	);
};

export default FormAlert;