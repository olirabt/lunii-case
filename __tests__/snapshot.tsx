import { render } from '@testing-library/react'
import FormAlert from "@/components/FormAlert";

describe("FormAlert", () => {
  it("renders the success message correctly", () => {
    const { container } = render(<FormAlert type="success" originalUrl="http://google.com/somelongparameter" shortUrl="https://shorturl.com" error={null} />)
    expect(container).toMatchSnapshot()
  });

  it("renders the warning message correctly", () => {
    const { container } = render(<FormAlert type="warning" originalUrl="google" shortUrl="" error="Lâ€™url saisie est invalide" />)
    expect(container).toMatchSnapshot()
  });
});