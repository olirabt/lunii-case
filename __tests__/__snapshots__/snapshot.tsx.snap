// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAlert renders the success message correctly 1`] = `
<div>
  <div
    class="text-left w-full px-5 py-3 mx-auto rounded-md bg-validate500 text-validate400"
  >
    <p>
      Voici le lien raccourci : 
      <a
        href="http://google.com/somelongparameter"
      >
        https://shorturl.com
      </a>
    </p>
  </div>
</div>
`;

exports[`FormAlert renders the warning message correctly 1`] = `
<div>
  <div
    class="text-left w-full px-5 py-3 mx-auto rounded-md bg-alert400 text-alert500"
  >
    <p>
      Lâ€™url saisie est invalide
    </p>
  </div>
</div>
`;
